### PR TITLE
Update node status when deployed by role

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

File: controllers/openstackdataplanerole_controller.go [hunk 1]
```diff
@@ -281,6 +281,14 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 		instance.Status.Deployed = true
 		r.Log.Info("Set ReadyCondition true")
 		instance.Status.Conditions.Set(condition.TrueCondition(condition.ReadyCondition, dataplanev1beta1.DataPlaneRoleReadyMessage))
+		for _, node := range nodes.Items {
+			node.Status.Conditions.MarkTrue(condition.ReadyCondition, fmt.Sprintf("Deployed by role.Name %s role.Namespace %s", instance.Name, instance.Namespace))
+			node.Status.Deployed = true
+			err = helper.PatchInstance(ctx, &node)
```
Comment (gibizer) [inline controllers/openstackdataplanerole_controller.go:287]: will this trigger a Reconcile on the OpenstackDataPlaneNode object? If so then that Reconcile might reset the status condition.

**Comment by fao89** [comment]
move to draft as I haven't tested yet

File: controllers/openstackdataplanerole_controller.go (hunk 1 above)
Comment (abays) [reply controllers/openstackdataplanerole_controller.go:287]: This [1] makes it seem like updates (patches too?) to status do trigger a reconcile, but it's from 2019.

[1] https://github.com/kubernetes-sigs/kubebuilder/issues/618

**Comment by fultonj** [comment]
FWIW I tested this patch and it worked for me.

**Review by slagle — CHANGES_REQUESTED** [review]

File: controllers/openstackdataplanerole_controller.go [hunk 2]
```diff
@@ -281,6 +281,14 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
...
+			if err != nil {
+				r.Log.Error(err, "Error on updating node status for ", node.Name)
```
[the remark this answers is not in the document]
Comment (fultonj) [reply controllers/openstackdataplanerole_controller.go:289]: I hit that too but didn't hit it after this change. 

Are you saying you have that and you still get that error? 

https://github.com/openstack-k8s-operators/dataplane-operator/compare/433d64926e0e37c48732055d434f37f4d65bf58d..b0737b7bd60f82f62f6bd6ce8952906d3e3da97e

[the remark this answers is not in the document]
Comment (fultonj) [reply controllers/openstackdataplanerole_controller.go:289]: I see. You hit a similar issue on a different line. 289 != 254. So this patch still needs that fix.

File: controllers/openstackdataplanenode_controller.go [hunk 3]
```diff
@@ -138,7 +138,8 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 	// persist any changes.
 	defer func() {
 		// update the overall status condition if service is ready
-		if instance.IsReady() {
+		if instance.IsReady() || instanceRole.IsReady() {
+			instance.Status.Deployed = true
```
Comment (slagle) [inline controllers/openstackdataplanenode_controller.go:142]: there's nothing to trigger a node reconcile so that this would get run

**Review by slagle — CHANGES_REQUESTED** [review]

File: controllers/openstackdataplanenode_controller.go (hunk 3 above)
Comment (slagle) [reply controllers/openstackdataplanenode_controller.go:142]: Yea, i think we need that loop back. This worked for me:

```
Mon Mar 27 01:55:14 (jslagle@teletran-1) [~/code/github/openstack-k8s-operators/dataplane-operator] (nodestatus)$ git diff
diff --git a/controllers/openstackdataplanerole_controller.go b/controllers/openstackdataplanerole_controller.go
index aa35f12..1af8bbf 100644
--- a/controllers/openstackdataplanerole_controller.go
+++ b/controllers/openstackdataplanerole_controller.go
@@ -282,6 +282,15 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
                r.Log.Info("Set ReadyCondition true")
                instance.Status.Conditions.Set(condition.TrueCondition(condition.ReadyCondition, dataplanev1beta1.DataPlaneRoleReadyMessage))
 
+               for _, node := range nodes.Items {
+                       _, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), &node, func() error {
+                               node.Status.Deployed = true
+                               return nil
+                       })
+                       if err != nil {
+                               return ctrl.Result{}, err
+                       }
+               }
                // Explicitly set instance.Spec.Deploy = false
                // We don't want another deploy triggered by any reconcile request, it should
                // only be triggered when the user (or another controller) specifically
```

Comment (fao89) [reply controllers/openstackdataplanenode_controller.go]: done!

**Comment by fao89** [comment]
CI fix: https://github.com/openstack-k8s-operators/dataplane-operator/pull/140

**Pull request merged**